### ISSUE TITLE
DXCreateInstance and DXCreateInstance2 not found when linking when _MSC_VER is defined

### ISSUE THREAD

**Issue by Kessica** [microsoft/DirectXShaderCompiler#3625]
dxcapi.h lines 86 and 95

#ifndef _MSC_VER
extern "C"
#endif
DXC_API_IMPORT HRESULT __stdcall DxcCreateInstance( ... );

When _MSC_VER is defined, e.g. when compiling using VS2019 (latest) "DxCreateInstance" and "DxCreateInstance2" exports are mangled, the linker cannot locate them. The exports definition file , "DXCompiler.def", contradicts the exports in the header file.  I believe the exports should all be extern "C" regardless of the _MSC_VER definition. This occurs when using the pre-built Windows set of libraries available as a download link

I am sorry if I haven't explained this very well, I am just a hobbyist programmer, trying to get things to link. Thanks.

**Comment by hekota** [comment]
Hello Kessica, I believe you are using an old version of dxcapi.h. The current dxcapi.h does not have the _MSC_VER define anymore, is has been removed 2.5 years ago. Are you getting this header from Windows SDK?

**Comment by Kessica** [comment]
Ahh, great suggestion. After checking, yes it is from WinSDK, despite specifying dxc\include first. In the end I just commented out the offending lines, but I have no idea what the consequences would be for other builds, hence just explaining my observation. I did attempt to follow the instructions to build the project files and use them, but even grabbing the latest cmake, I couldnt get past all the errors, so ended up just trying make it all hang together. I think vs2019 was the cause using an old version of cmake, but documentation on changing vs2019 to use the new version is exceptionally sparse. I dont want to create all the projects by hand, so perhaps that is why the paths are not set up properly. Thanks for a fast response.
